pkg/cluster/operation/scale_in.go

```diff
@@ -122,29 +124,68 @@ func ScaleInCluster(
 		}
 	}
 
+	// At least a PD server exists
+	if len(pdEndpoint) == 0 {
+		return errors.New("cannot find available PD instance")
+	}
+
+	pdClient := api.NewPDClient(pdEndpoint, 10*time.Second, tlsCfg)
+
+	binlogClient, err := api.NewBinlogClient(pdEndpoint, tlsCfg)
+	if err != nil {
+		return err
+	}
+
+	offlineInstance := func(component spec.Component, instance spec.Instance) error {
+		ignoreErr := options.Force
+
+		// just try to stop and destroy
+		if err := StopComponent(getter, []spec.Instance{instance}, options.OptTimeout); err != nil {
+			if !ignoreErr {
+				return errors.Annotatef(err, "failed to stop %s", component.Name())
+			}
+			log.Warnf("failed to stop %s: %v", component.Name(), err)
+		}
+		if err := DestroyComponent(getter, []spec.Instance{instance}, cluster, options); err != nil {
+			if !ignoreErr {
+				return errors.Annotatef(err, "failed to destroy %s", component.Name())
+			}
+			log.Warnf("failed to destroy %s: %v", component.Name(), err)
+		}
+
+		monitoredOptions := cluster.GetMonitoredOptions()
+		instCount[instance.GetHost()]--
+		if instCount[instance.GetHost()] == 0 && cluster.GetMonitoredOptions() != nil {
```

Comment:
cluster.GetMonitoredOptions() != nil is not needed because it's always true here: https://github.com/pingcap/tiup/blob/master/pkg/cluster/spec/spec.go#L192

Reply:
Sorry, this logical is forked from https://github.com/pingcap/tiup/blob/7aa4ea40705925f32e3033854e3bb786e87395e9/pkg/cluster/operation/destroy.go#L75-L79

In which case, the cluster may be tidb or dm, don't think too much, just copied it out.